lib/console.js

```diff
@@ -35,20 +37,40 @@ function Console(stdout, stderr) {
   }
 }
 
+function write(ignoreErrors, stream, string) {
+  if (!ignoreErrors) return stream.write(string);
+
+  // There may be an error occurring synchronously (e.g. for files or TTYs)
+  // or asynchronously (e.g. pipes), so handle both situations.
+  try {
+    stream.write(string, (err) => {
+      if (err) {
+        // If there was an error, it will be emitted on `stream` as
+        // an `error` event. Adding a `once` listener will keep that error
+        // from becoming an uncaught exception, but since the handler is
+        // removed after the event, non-console.* writes won’t be affected.
+        stream.once('error', noop);
```

Comment:
@addaleax, technically @jasnell has a point. You will need to wrap `write` in a `once` and `removeListener`pair : in https://github.com/nodejs/node/blob/master/lib/_stream_writable.js#L340-L346, if `sync`  is true the error is emitted before, then after.